packages/protocol-kit/src/Safe.ts

```diff
@@ -562,21 +680,47 @@ class Safe {
     if (!addressIsOwner) {
       throw new Error('Transactions can only be signed by Safe owners')
     }
-
+    const isContractSignature = signingMethod === SigningMethod.SAFE_SIGNATURE
+    const method = isContractSignature ? 'eth_sign' : signingMethod
     let signature: SafeSignature
-    if (signingMethod === 'eth_signTypedData_v4') {
+
+    if (method === 'eth_signTypedData_v4') {
       signature = await this.signTypedData(transaction, 'v4')
-    } else if (signingMethod === 'eth_signTypedData_v3') {
+    } else if (method === 'eth_signTypedData_v3') {
       signature = await this.signTypedData(transaction, 'v3')
-    } else if (signingMethod === 'eth_signTypedData') {
-      signature = await this.signTypedData(transaction)
+    } else if (method === 'eth_signTypedData') {
```

Comment:
Another example of a comparison as string of values that exist on SigningMethod enum, that is the default value used in the function header.